### ISSUE TITLE
Tests fail on Ubuntu 14.04 (32-bit)

### ISSUE THREAD

**Issue by henriqueesteves** [jnr/jnr-ffi#34]
When running tests on Ubuntu 14.04 (32-bit), I get the following result:

```
 Failed tests: 
  mixObjectsAndPrimitives(jnr.ffi.PointerTest)
  s8Array(jnr.ffi.struct.ArrayTest)

Tests in error: 
  shortByReference(jnr.ffi.ArrayTest)
  doubleByReference(jnr.ffi.ArrayTest)
  setLong32ByReference(jnr.ffi.ArrayTest)
  setLongByReference(jnr.ffi.ArrayTest)
  floatByReference(jnr.ffi.ArrayTest)
  intByReference(jnr.ffi.ArrayTest)
  byteByReference(jnr.ffi.ArrayTest)
  getLongByReference(jnr.ffi.ArrayTest)
  jnr.ffi.BufferTest

Tests run: 232, Failures: 2, Errors: 9, Skipped: 0
```

All "Tests in error" fail for the same reason:

```java
java.lang.IllegalArgumentException: unsupported type class [J
	at jnr.ffi.provider.jffi.X86MethodGenerator.generateWrapper(X86MethodGenerator.java:148)
	at jnr.ffi.provider.jffi.X86MethodGenerator.generate(X86MethodGenerator.java:108)
	at jnr.ffi.provider.jffi.AsmLibraryLoader.generateInterfaceImpl(AsmLibraryLoader.java:132)
	at jnr.ffi.provider.jffi.AsmLibraryLoader.loadLibrary(AsmLibraryLoader.java:59)
	at jnr.ffi.provider.jffi.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:43)
	at jnr.ffi.LibraryLoader.load(LibraryLoader.java:269)
	at jnr.ffi.TstUtil.loadTestLib(TstUtil.java:38)
	at jnr.ffi.TstUtil.loadTestLib(TstUtil.java:28)
	at jnr.ffi.ArrayTest.setUpClass(ArrayTest.java:96)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:62)
	at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.executeTestSet(AbstractDirectoryTestSuite.java:140)
	at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.execute(AbstractDirectoryTestSuite.java:127)
	at org.apache.maven.surefire.Surefire.run(Surefire.java:177)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:338)
	at org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:997)
```

If I run the tests with the option "jnr.ffi.x86asm.enabled" disabled, I get the same result, but the exception changes to:

```java
java.lang.IllegalArgumentException: unsupported parameter type class [J
	at jnr.ffi.provider.jffi.AbstractFastNumericMethodGenerator.generate(AbstractFastNumericMethodGenerator.java:47)
	at jnr.ffi.provider.jffi.BaseMethodGenerator.generate(BaseMethodGenerator.java:42)
	at jnr.ffi.provider.jffi.AsmLibraryLoader.generateInterfaceImpl(AsmLibraryLoader.java:132)
	at jnr.ffi.provider.jffi.AsmLibraryLoader.loadLibrary(AsmLibraryLoader.java:59)
	at jnr.ffi.provider.jffi.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:43)
	at jnr.ffi.LibraryLoader.load(LibraryLoader.java:269)
	at jnr.ffi.TstUtil.loadTestLib(TstUtil.java:38)
	at jnr.ffi.TstUtil.loadTestLib(TstUtil.java:28)
	at jnr.ffi.ArrayTest.setUpClass(ArrayTest.java:96)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:62)
	at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.executeTestSet(AbstractDirectoryTestSuite.java:140)
	at org.apache.maven.surefire.suite.AbstractDirectoryTestSuite.execute(AbstractDirectoryTestSuite.java:127)
	at org.apache.maven.surefire.Surefire.run(Surefire.java:177)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.maven.surefire.booter.SurefireBooter.runSuitesInProcess(SurefireBooter.java:338)
	at org.apache.maven.surefire.booter.SurefireBooter.main(SurefireBooter.java:997)
```
All "Failed tests" fail for a possible bug in "StructArrayParameterConverter.java ". I have a possible solution (https://github.com/henriqueesteves/jnr-ffi/commit/d43ac0e021eb1d45311d9bea8979197781b4a719), however, I have not submitted a pull request because I would like that all tests pass.

**Comment by headius** [comment]
Sounds good, we'll look forward to the PR.

**Comment by headius** [comment]
Or perhaps I misunderstood...your fix is for the reported issue, yes? And you want to make sure it passes tests, or it does not pass tests?

**Comment by henriqueesteves** [comment]
My fix corrects only the "Failed tests". I found this bug in a library that I develop and realized that the "Failed tests" have exposed the same bug.

**Comment by henriqueesteves** [comment]
I opened a pull request (https://github.com/jnr/jnr-ffi/pull/35) that corrects only the "Failed tests".

**Comment by headius** [comment]
Oh, I see! So basically there are a number of tests not passing on Ubuntu 14.04 32-bit, but you tracked down some of those failures to this fix. Thank you for clarifying! Are you planning on trying to get the other tests passing too?

**Comment by henriqueesteves** [comment]
@headius I opened a pull request (https://github.com/jnr/jnr-ffi/pull/38) that fix all tests.

**Comment by headius** [comment]
Fixed by #38.
